### PR TITLE
Enchanted armoire test improvements

### PR DESCRIPTION
**Pull request by crookedneighbor**
@SabreCat please review
## Problem

We had to update the armoire test every time a new item was added
## Solution

Define a full armoire programmatically based on Content file
## Notes
- Moved the armoire purchase tests to a separate file 
- Converted the tests to javascript (to make the eventual migration to es2015 easier)
- Added a gulp task for watching the common test

### CODE REVIEW

File: test/common/armoire.js [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+
+    sinon.stub(user.fns, 'predictableRandom');
+  });
+
+  afterEach(function() {
+    user.fns.randomVal.restore();
+    user.fns.predictableRandom.restore();
+  });
+
+  context('Potion', function() {
```
Comment (SabreCat) [inline test/common/armoire.js]: These probably don't need to stick around in an Armoire-specific test file.

File: test/common/armoire.js [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+      expect(user.items.food).to.eql({'Honey': 1});
+      expect(user.stats.exp).to.eql(0);
+      expect(user.stats.gp).to.eql(100);
+    });
+
+    it('gives more equipment', function() {
+      user.fns.predictableRandom.returns(YIELD_EQUIPMENT);
+      user.items.gear.owned = {
+        weapon_warrior_0: true,
+        head_armoire_hornedIronHelm: true
```
Comment (SabreCat) [inline test/common/armoire.js]: This is another piece I've had to edit manually, if we can figure out how to correctly predict what the new equipment yield should be when available items change.

File: test/common/armoire.js [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+    var YIELD_EQUIPMENT = .5;
+    var YIELD_FOOD = .7;
+    var YIELD_EXP = .9;
+
+    var fullArmoire = {}
+
+    _(Content.gearTypes).each(function(type) {
+      _(Content.gear.tree[type].armoire).each(function(gearObject, gearName) {
+        armoireKey = gearObject.key;
+        fullArmoire[armoireKey] = true;
```
Comment (SabreCat) [inline test/common/armoire.js]: Yay!

File: test/common/armoire.js (hunk 1 above)
Comment (crookedneighbor) [inline test/common/armoire.js]: This is going to be the user.fns.buy test file

Comment (crookedneighbor) [inline test/common/armoire.js]: Whoops. I never renamed it.

**Pull request merged**